Sources/SmokeAWSHttp/AWSClientInvocationTraceContext.swift

```diff
@@ -70,49 +70,51 @@ public struct AWSClientInvocationTraceContext: InvocationTraceContext {
             return
         }
         
-        let logLine = getLogLine(successfullyCompletedRequest: false, response: response, bodyData: bodyData, error: error)
-        
         // log at error if this is a server error
+        let level: Logger.Level
         if let response = response, response.status.code >= 500 && response.status.code < 600 {
```

Comment:
Should we also log 4xx responses at error level?

Reply:
This potentially could give a lot of false positive errors in the logs that may not actually be an issue. For example `ConditionalCheckFailedException` from DynamoDb returns as a 400 error but the application code often will be expecting this and will handle the error as part of its application logic.

https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_PutItem.html

Reply:
It's true that 4xx errors are sometimes expected in the client-server flow. On the other hand, some are actual errors too (for example InvalidRequest). I guess this comes down to the service design and whether it uses success or 4xx responses to signal a valid but unexpected state.

I'm a bit on the fence on this one, but if the goal is to streamline the standard logs then it's probably best to avoid logging 4xx responses at error level.

Could this be configurable in the `AWSClientInvocationTraceContext` object with some sort of `log4xxResponsesAsError` flag (defaulting to false)?

Reply:
Having a configuration flag seems like a reasonable approach. I will create a follow up PR to add this.